Prometheus.HttpExporter.AspNetCore/HttpRequestMiddlewareBase.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Prometheus.Advanced;
+
+namespace Prometheus.HttpExporter.AspNetCore
+{
+    public class HttpRequestMiddlewareBase<T> where T : ICollector
+    {
+        public HttpRequestMiddlewareBase(T collector)
+        {
+            if (collector == null || !this.LabelsAreValid(collector)) throw new ArgumentException(nameof(collector)); 
+            this._labelNames = collector.LabelNames;
+            this._labelData = this._labelNames.ToDictionary(key => key);
+            this._requiresRouteData = this._labelData.ContainsKey(LabelNames.Action) ||
+                                      this._labelData.ContainsKey(LabelNames.Controller);
+        }
+
+        protected string[] GetLabelData(HttpContext context)
+        {
+            if (this._labelNames.Length == 0)
+            {
+                return new string[0];
+            }
+
+            if (this._requiresRouteData)
+            {
+                var routeData = context.GetRouteData();
+
+                this.UpdateMetricValueIfExists(LabelNames.Method, context.Request.Method);
+                this.UpdateMetricValueIfExists(LabelNames.Code, context.Response.StatusCode.ToString());
+                this.UpdateMetricValueIfExists(LabelNames.Action, routeData?.Values["Action"] as string ?? string.Empty);
+                this.UpdateMetricValueIfExists(LabelNames.Controller, routeData?.Values["Controller"] as string ?? string.Empty);
+            }
+            else
+            {
+                this.UpdateMetricValueIfExists(LabelNames.Method, context.Request.Method);
+                this.UpdateMetricValueIfExists(LabelNames.Code, context.Response.StatusCode.ToString());
```

Comment:
These two lines appear to be duplicated on both sides of the if/else. Unless I miss something obvious, it seems that they can be moved outside and de-duplicated.